### PR TITLE
CConfigFileMemory unittest

### PR DESCRIPTION
**Pull request by GYengera**
**Changed apps/libraries: Base Utils** 
*  Unit test for CConfigFileMemory

I acknowledge to have: 
* Read the [`CONTRIBUTING`](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md) page
* Updated [`docs/doxygen-pages/changelog.h`](https://github.com/MRPT/mrpt/blob/master/doc/doxygen-pages/changeLog_doc.h) to describe these changes (if applicable)
* Updated [`AUTHORS`](https://github.com/MRPT/mrpt/blob/master/AUTHORS) (if applicable)

(Notify: @MRPT/owners )

### CODE REVIEW

File: libs/base/src/utils/CConfigFileMemory_unittest.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   |                                                                           |
+   | Copyright (c) 2005-2017, Individual contributors, see AUTHORS file        |
+   | See: http://www.mrpt.org/Authors - All rights reserved.                   |
+   | Released under BSD License. See details in http://www.mrpt.org/License    |
+   +---------------------------------------------------------------------------+ */
+
+#include <mrpt/utils/CConfigFileMemory.h>
+#include <mrpt/system/os.h>
+
+#include "simpleini/SimpleIni.h"
```
Comment (jlblancoc) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: I think this #include is not required, right? Please, remove it...

File: libs/base/src/utils/CConfigFileMemory_unittest.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+#include <gtest/gtest.h>
+
+using namespace mrpt;
+using namespace mrpt::utils;
+using namespace mrpt::utils::simpleini;
```
Comment (jlblancoc) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: Idem... not needed from outside of the CConfig* classes... "Users" are not expected to even know about this namespace ;-)

File: libs/base/src/utils/CConfigFileMemory_unittest.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+using namespace std;
+
+class TConfig : public CConfigFileMemory {
+	public:
+	using CConfigFileMemory::writeString;
+	using CConfigFileMemory::readString;
+	using CConfigFileMemory::getAllSections;
+	using CConfigFileMemory::getAllKeys;
+	};
```
Comment (jlblancoc) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: I don't understand the motivation of this class here... Why did you need to create it? 
Why not directly define the test objects of type "CConfigFileMemory" instead of TConfig?

File: libs/base/src/utils/CConfigFileMemory_unittest.cpp [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+{
+	vector_string names;
+	TConfig fourth;
+	fourth.writeString("sec","name","val");
+	fourth.writeString("sec","names","value");
+	fourth.getAllKeys("sec", names);
+	EXPECT_STREQ("name",names[0].c_str());
+	EXPECT_STREQ("names",names[1].c_str());
+	EXPECT_EQ(2,names.size());
+}
```
Comment (jlblancoc) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: I would love to see a test, testing the capability of initializing the CConfigFileMemory from a multiline string containing example entries. But I could add it myself after accepting this PR...

File: libs/base/src/utils/CConfigFileMemory_unittest.cpp (hunk 3 above)
Comment (GYengera) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: The functions within CConfigFileMemory are protected, so the only way I knew of accessing them was through an inherited class. If there is a better way then please do let me know.

Comment (jlblancoc) [inline libs/base/src/utils/CConfigFileMemory_unittest.cpp]: ah, I see!

I just added [this commit](https://github.com/MRPT/mrpt/commit/b6dd211cffd73ec09af01e54321d2e87aadd6f1d) which hopefully will solve your confusion. Those writeString() & readString() are *virtual* methods, not intended for usage from the user's side. 

getAllSections() and getAllKeys() are public, you should have no problem directly using them...

**Comment by jlblancoc** [comment]
👍 

Please, see my further changes in https://github.com/MRPT/mrpt/commit/cd06d5c90ae06bd99f00548884188d6472aa03cf

(You'll have to re-fork or do a PR towards your fork to synch). 

The new feature originally planned in #434 is the one marked as a "TO DO" by the bottom of the file.
